### PR TITLE
experiments_submissions_backend

### PR DESCRIPTION
**Pull request by nooraangelva**
Ref: https://github.com/cern-sis/issues-inspire/issues/18

### CODE REVIEW

File: backend/inspirehep/submissions/views.py [hunk 1]
```diff
@@ -259,12 +260,23 @@ def create_ticket(self, record, rt_template):
         )
 
 
+class ExperimentSubmissionsResource(BaseSubmissionsResource):
+    def post(self):
+        """Adds new experiment record"""
+
+        data = self.load_data_from_request()["data"]
```
Comment (MJedr) [inline backend/inspirehep/submissions/views.py:267]: what if there'll be key error?

I would suggest to create a dict with required fields and then update it with data from request.

File: backend/inspirehep/submissions/views.py [hunk 2]
```diff
@@ -259,12 +260,23 @@ def create_ticket(self, record, rt_template):
...
+        data["_collections"] = ["Experiments"]
+        data["$schema"] = "https://labs.inspirehep.net/schemas/records/experiments.json"
+        record = ExperimentsRecord(data=data).create(data)
+        db.session.commit()
+        return (record, 201)
```
Comment (MJedr) [inline backend/inspirehep/submissions/views.py:272]: are you sure we want to return a tuple of record and int 201? :p 
I think that you want to return `control_number` and status code using `jsonify`

File: backend/tests/integration/submissions/test_submissions_views.py [hunk 3]
```diff
@@ -2152,3 +2153,48 @@ def test_new_seminar_submission_with_invalid_lit_record(
...
+
+    payload = orjson.loads(response.data)
+    experiment_id = payload["control_number"]
+    experiment_record = ExperimentsRecord.get_record_by_pid_value(experiment_id)
+
+    experiment_record_data = {
+        key: value
+        for (key, value) in experiment_record.items()
+        if key in REQUIRED_EXPERIMENT_RECORD_DATA
+    }
```
Comment (MJedr) [inline backend/tests/integration/submissions/test_submissions_views.py:2199]: can't you just check if experiment record data equals expected data?

File: backend/inspirehep/submissions/views.py [hunk 4]
```diff
@@ -259,12 +263,31 @@ def create_ticket(self, record, rt_template):
...
+
+    def post(self):
+        """Adds new experiment record"""
+
+        request_data = self.load_data_from_request()
+        data = {
+            "_collections": ["Experiments"],
+            "$schema": "https://inspirehep.net/schemas/records/experiments.json",
+            "legacy_name": request_data["legacy_name"],
+            "project_type": request_data["project_type"],
```
Comment (drjova) [inline backend/inspirehep/submissions/views.py:278]: they key might be missing 🤔

File: backend/inspirehep/submissions/views.py [hunk 5]
```diff
@@ -259,12 +263,31 @@ def create_ticket(self, record, rt_template):
...
+    def load_data_from_request(self):
+        return experiment_loader_v1()
+
+    def post(self):
+        """Adds new experiment record"""
+
+        request_data = self.load_data_from_request()
+        data = {
+            "_collections": ["Experiments"],
+            "$schema": "https://inspirehep.net/schemas/records/experiments.json",
```
Comment (drjova) [inline backend/inspirehep/submissions/views.py:276]:
```suggestion
            "$schema":  url_for(
            "invenio_jsonschemas.get_schema",
            schema_path="records/experiments.json",
            _external=True,
        )
```

File: backend/tests/integration/submissions/test_submissions_views.py [hunk 6]
```diff
@@ -2152,3 +2153,48 @@ def test_new_seminar_submission_with_invalid_lit_record(
...
+}
+
+
+@pytest.mark.parametrize(
+    "form_data,expected_record_data",
+    [
+        (deepcopy(EXPERIMENT_FORM_DATA), REQUIRED_EXPERIMENT_RECORD_DATA),
+    ],
+)
+def test_new_experiment_submission(
```
Comment (drjova) [inline backend/tests/integration/submissions/test_submissions_views.py:2177]: we should have a test case with sending empty data to the API

File: backend/inspirehep/submissions/views.py (hunk 5 above)
Comment (MJedr) [inline backend/inspirehep/submissions/views.py:276]: I would put that in marshmallow, also if you'll have a proper serializer `self.load_data_from_request()` should prepare your data in a correct format for creating a record.

File: backend/inspirehep/submissions/views.py [hunk 7]
```diff
@@ -259,12 +263,31 @@ def create_ticket(self, record, rt_template):
...
+            "legacy_name": request_data["legacy_name"],
+            "project_type": request_data["project_type"],
+        }
+        record = ExperimentsRecord(data=data).create(data)
+        db.session.commit()
+        experiment_serializer_v1.dump(record)
```
Comment (MJedr) [inline backend/inspirehep/submissions/views.py:282]: why you need it?

File: backend/inspirehep/submissions/views.py (hunk 5 above)
Comment (nooraangelva) [reply backend/inspirehep/submissions/views.py:276]: hmm, but how do I add the ```$schema``` there? $ makes it a bit tricky or do you mean the whole dict?

Comment (MJedr) [reply backend/inspirehep/submissions/views.py:276]: you can have a method for building this data dict (which won't use builder).

Comment (nooraangelva) [reply backend/inspirehep/submissions/views.py:276]: Ah okay. Now I understand. Thank you

File: backend/inspirehep/submissions/views.py (hunk 7 above)
Comment (nooraangelva) [reply backend/inspirehep/submissions/views.py:282]: Don't I need to serialize the record or is it okay without serialization? On the other hand, I'm returning just the control number and status.

**Review by MJedr — APPROVED** [review]

**Pull request merged**